kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTaskV2.java

```diff
@@ -40,10 +45,22 @@ public void put(Collection<SinkRecord> records) throws ConnectException {
         if(transformedRecords.isEmpty()){
             return;
         }
-        String eventNameTag = String.format(EVENT_NAME_TAG, transformedRecords.get(0).topic());
+        SinkRecord firstRecord = transformedRecords.get(0);
+        long eventReceivedTimeMs = 0L;
+        try {
+            Map<String, Object> eventPayload = (Map<String, Object>) firstRecord.value();
+            Map<String, Object> properties = (Map<String, Object>) eventPayload.get("properties");
+            eventReceivedTimeMs = (Long) properties.get("ts_received_ms");
+            // log.warn("ts_received_ms: " + eventReceivedTimeMs);
+        } catch (ClassCastException | NullPointerException ex) {
+            // Skip the record if it doesn't have the required fields, do nothing
+            // log.warn("Skipping the record as it doesn't have the required fields", ex);
+        }
+        String eventNameTag = String.format(EVENT_NAME_TAG, firstRecord.topic());
         long putStartTime = System.nanoTime();
         super.put(transformedRecords);
         statsDClient.timing("put.time", System.nanoTime()-putStartTime, eventNameTag);
+        statsDClient.gauge("batch.ts_received_ms", eventReceivedTimeMs, eventNameTag);
```

Comment:
why are we sending first ts_received_ms from the batch? didn't we want to send the minimum ts_received_ms across entire batch?

Reply:
disscussed on slack